syncopy/tests/test_connectivity.py

```diff
@@ -0,0 +1,584 @@
+        assert cross_02[lag_idx] > 0.99
+        # same for a period multiple
+        assert cross_02[lag_idx + period_idx] > .99
+        # plus half the period a minimum occurs
+        assert cross_02[lag_idx + period_idx // 2] < -.99
+
+        # test for (anti-)symmetry
+        cross_20 = corr.data[:, 0, 2, 0]
+        assert cross_20[-lag_idx] > 0.99
+        assert cross_20[-lag_idx - period_idx] > 0.99
+
+        # only plot for simple solution test
+        if len(kwargs) == 0:
+            plot_corr(corr, 0, 0, label='corr 0-0')
+            plot_corr(corr, 1, 1, label='corr 1-1')
+            plot_corr(corr, 0, 2, label='corr 0-2')
+            ppl.xlim((-.01, 0.5))
+            ppl.ylim((-1.1, 1.3))
+            ppl.legend(ncol=3)
+
+    def test_corr_padding(self):
+
+        self.test_corr_solution(pad_to_length=None)
+        # no padding is allowed for
+        # this method        
+        try:
+            self.test_corr_solution(pad_to_length=1000)
+        except SPYValueError as err:
+            assert 'pad_to_length' in str(err)
+            assert 'no padding needed/allowed' in str(err)
+            
+        try:
+            self.test_corr_solution(pad_to_length='nextpow2')
+        except SPYValueError as err:
+            assert 'pad_to_length' in str(err)
+            assert 'no padding needed/allowed' in str(err)
+
+        try:
+            self.test_corr_solution(pad_to_length='IamNoPad')
+        except SPYValueError as err:
+            assert 'Invalid value of `pad_to_length`' in str(err)
+            assert 'nextpow2' in str(err)
+
+    def test_corr_selections(self):
+
+        selections = mk_selection_comb(self.nTrials,
+                                       self.nChannels,
+                                       *self.time_span)
+
+        for sel_dct in selections:
+
+            result = connectivity(self.data, method='corr', select=sel_dct)
+
+            # check here just for finiteness and positivity
+            assert np.all(np.isfinite(result.data))
+
+    def test_corr_cfg(self):
+
+        call = lambda cfg: connectivity(self.data, cfg)
+        run_cfg_test(call, method='corr', positivity=False)
+
+    @skip_without_acme
+    @skip_low_mem
+    def test_corr_parallel(self, testcluster=None):
+
+        ppl.ioff()
+        client = dd.Client(testcluster)
+        all_tests = [attr for attr in self.__dir__()
+                     if (inspect.ismethod(getattr(self, attr)) and 'parallel' not in attr)]
+
+        for test in all_tests:
+            test_method = getattr(self, test)
+            test_method()
+        client.close()
+        ppl.ion()
+
+
+# -- helper functions --
+
+
+def run_padding_test(call, pad_length):
+    """
+    The callable should test a solution and support
+    a single keyword argument `pad_to_length`
+    """
+
+    pad_options = [pad_length, 'nextpow2', None]
+    for pad in pad_options:
+        call(pad_to_length=pad)
+
+    # test invalid pads
+    try:
```

Comment:
same as below - `pytest` offers a context manager for these types of exception tests